### DISCUSSION TITLE
keyboard layout handling (having to switch between english and spanish frequently)

### DISCUSSION THREAD

**Discussion by tokisuno** [kovidgoyal/kitty#6252] [Q&A] [answered]
howdy y'all

basically i write in vim. part of this will involve me writing my thesis in vim which involves analysis in both english and spanish, meaning i have to switch between both layouts. the issue with this is kitty doesn't seem to take layout switching very well. i can't type in japanese (not really an issue for me but sometimes annoying), and i can't switch my ibus layout without going over to firefox, switching my layout there, typing in the search bar to make sure it works, and then going back into kitty. this is very annoying and tedious for me. is there way around this, or any fix?

**Comment by kovidgoyal** [comment]
Switching of ibus layouts has nothing to do with kitty, it happens in IBUS. I dont see what kitty could do to prevent you switching layouts.

**Reply by tokisuno** [reply]
i don't either. kitty is the only terminal i've used that has ibus issues. hell, urxvt functions better with it.

**Reply by kovidgoyal** [reply]
ibus works fine for me with kitty. Switching, japanese input, entering emoji they all work.

**Reply by tokisuno** [reply]
really? it hasn't worked on any computer i've used... that's all without tweaking right? what version of ibus are you on?

**Reply by kovidgoyal** [reply, marked as answer]
You need to set the GLFW_IM_MODULE=ibus env var for kitty to even use IBUS.

**Reply by tokisuno** [reply]
forgot to mark as answered. my bad
